inference-engine/tests/functional/inference_engine/transformations/compare_functions_test.cpp

```diff
@@ -0,0 +1,211 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include "common_test_utils/test_common.hpp"
+#include <string>
+#include <memory>
+#include <queue>
+
+#include <ngraph/pass/manager.hpp>
+#include <ngraph/function.hpp>
+#include <ngraph/opsets/opset5.hpp>
+#include <transformations/utils/utils.hpp>
+#include <transformations/init_node_info.hpp>
+
+#include "common_test_utils/ngraph_test_utils.hpp"
+
+using namespace testing;
+using namespace ngraph;
+
+TEST(TransformationTests, CompareFunctoinsTIPositive) {
+    std::shared_ptr<ngraph::Function> f(nullptr), f_ref(nullptr);
```

Comment:
IMHO, we should use [IIFE](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-lambda-init) for such complex initialization.

Reply:
Yes, I agree. I'll do it in a separate PR not to block merge of this one.